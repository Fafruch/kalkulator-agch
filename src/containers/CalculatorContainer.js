import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import Calculator from '../components/Calculator/Calculator'
import subjectsPicker from '../store/selectors/subjectsPicker'
import { toggleScoreTable } from '../actions/index'

const mapStateToProps = (state) => ({
  maxScoreWrapper: subjectsPicker(state.subjects),
  isScoreTableOpened: state.scoreTable.isOpened,
  isAnySubjectPresent: !!state.subjects.primary.length || !!state.subjects.lingual.length
})

const mapDispatchToProps = (dispatch) => ({
  onToggleClick: bindActionCreators(toggleScoreTable, dispatch)
})

const CalculatorContainer = connect(mapStateToProps, mapDispatchToProps)(Calculator)

export default CalculatorContainer
