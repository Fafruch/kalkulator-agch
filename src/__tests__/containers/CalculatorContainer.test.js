import React from 'react'
import { createStore } from 'redux'
import { shallow } from 'enzyme'

import rootReducer from '../../store/reducers'
import CalculatorContainer from '../../containers/CalculatorContainer'

const setup = (setupProps = {}) => {
  const defaultProps = { store: createStore(rootReducer) }
  const props = { ...defaultProps, ...setupProps }
  const wrapper = shallow(<CalculatorContainer {...props} />)

  return {
    props,
    wrapper
  }
}

describe('CalculatorContainer component', () => {
  test('should render', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })
})
