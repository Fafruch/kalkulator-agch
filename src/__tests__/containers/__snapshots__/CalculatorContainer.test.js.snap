// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorContainer component should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Calculator
    isAnySubjectPresent={false}
    isScoreTableOpened={false}
    maxScoreWrapper={
        Object {
            "lingual": Object {
              "maxScore": Object {
                "formula": "",
                "value": 0,
              },
              "name": "",
            },
            "primary": Object {
              "maxScore": Object {
                "formula": "",
                "value": 0,
              },
              "name": "",
            },
          }
    }
    onToggleClick={[Function]}
    store={
        Object {
            "asyncReducers": Object {},
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "notify": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "asyncReducers": Object {},
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "unsubscribe": null,
          }
    }
/>,
  "nodes": Array [
    <Calculator
      isAnySubjectPresent={false}
      isScoreTableOpened={false}
      maxScoreWrapper={
            Object {
                  "lingual": Object {
                    "maxScore": Object {
                      "formula": "",
                      "value": 0,
                    },
                    "name": "",
                  },
                  "primary": Object {
                    "maxScore": Object {
                      "formula": "",
                      "value": 0,
                    },
                    "name": "",
                  },
                }
      }
      onToggleClick={[Function]}
      store={
            Object {
                  "asyncReducers": Object {},
                  "dispatch": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                  Symbol(observable): [Function],
                }
      }
      storeSubscription={
            Subscription {
                  "listeners": Object {
                    "notify": [Function],
                  },
                  "onStateChange": [Function],
                  "parentSub": undefined,
                  "store": Object {
                    "asyncReducers": Object {},
                    "dispatch": [Function],
                    "getState": [Function],
                    "replaceReducer": [Function],
                    "subscribe": [Function],
                    Symbol(observable): [Function],
                  },
                  "unsubscribe": null,
                }
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Connect(Calculator)
        store={
                Object {
                        "asyncReducers": Object {},
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Connect {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "store": undefined,
          "storeSubscription": undefined,
        },
        "notifyNestedSubs": [Function],
        "props": Object {
          "store": Object {
            "asyncReducers": Object {},
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "propsMode": true,
        "refs": Object {},
        "renderCount": 0,
        "selector": Object {
          "error": null,
          "props": Object {
            "isAnySubjectPresent": false,
            "isScoreTableOpened": false,
            "maxScoreWrapper": Object {
              "lingual": Object {
                "maxScore": Object {
                  "formula": "",
                  "value": 0,
                },
                "name": "",
              },
              "primary": Object {
                "maxScore": Object {
                  "formula": "",
                  "value": 0,
                },
                "name": "",
              },
            },
            "onToggleClick": [Function],
            "store": Object {
              "asyncReducers": Object {},
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
          },
          "run": [Function],
          "shouldComponentUpdate": false,
        },
        "setWrappedInstance": [Function],
        "state": Object {},
        "store": Object {
          "asyncReducers": Object {},
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "subscription": Subscription {
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "asyncReducers": Object {},
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
        "version": 1,
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Calculator
          isAnySubjectPresent={false}
          isScoreTableOpened={false}
          maxScoreWrapper={
                    Object {
                              "lingual": Object {
                                "maxScore": Object {
                                  "formula": "",
                                  "value": 0,
                                },
                                "name": "",
                              },
                              "primary": Object {
                                "maxScore": Object {
                                  "formula": "",
                                  "value": 0,
                                },
                                "name": "",
                              },
                            }
          }
          onToggleClick={[Function]}
          store={
                    Object {
                              "asyncReducers": Object {},
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
          storeSubscription={
                    Subscription {
                              "listeners": Object {
                                "notify": [Function],
                              },
                              "onStateChange": [Function],
                              "parentSub": undefined,
                              "store": Object {
                                "asyncReducers": Object {},
                                "dispatch": [Function],
                                "getState": [Function],
                                "replaceReducer": [Function],
                                "subscribe": [Function],
                                Symbol(observable): [Function],
                              },
                              "unsubscribe": null,
                            }
          }
/>,
        "_debugID": 2,
        "_renderedOutput": <Calculator
          isAnySubjectPresent={false}
          isScoreTableOpened={false}
          maxScoreWrapper={
                    Object {
                              "lingual": Object {
                                "maxScore": Object {
                                  "formula": "",
                                  "value": 0,
                                },
                                "name": "",
                              },
                              "primary": Object {
                                "maxScore": Object {
                                  "formula": "",
                                  "value": 0,
                                },
                                "name": "",
                              },
                            }
          }
          onToggleClick={[Function]}
          store={
                    Object {
                              "asyncReducers": Object {},
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
          storeSubscription={
                    Subscription {
                              "listeners": Object {
                                "notify": [Function],
                              },
                              "onStateChange": [Function],
                              "parentSub": undefined,
                              "store": Object {
                                "asyncReducers": Object {},
                                "dispatch": [Function],
                                "getState": [Function],
                                "replaceReducer": [Function],
                                "subscribe": [Function],
                                Symbol(observable): [Function],
                              },
                              "unsubscribe": null,
                            }
          }
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Connect(Calculator)
    store={
        Object {
            "asyncReducers": Object {},
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
/>,
}
`;
